variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/factories/VariantVcfExacFactory.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2014-2016 EMBL - European Bioinformatics Institute
+ * Copyright 2015 OpenCB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.commons.core.models.factories;
+
+import uk.ac.ebi.eva.commons.core.models.VariantStatistics;
+import uk.ac.ebi.eva.commons.core.models.genotype.Genotype;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.VariantSourceEntry;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Overrides the methods in VariantAggregatedVcfFactory that take care of the fields QUAL, FILTER and INFO, to support
+ * the specific format of Exome Aggregation Consortium VCFs.
+ */
+public class VariantVcfExacFactory extends VariantAggregatedVcfFactory {
+
+    private static final String AC_HOM = "AC_Hom";
+
+    private static final String AC_HET = "AC_Het";
+
+    private static final String AN_ADJ = "AN_Adj";
+
+    private static final String AC_ADJ = "AC_Adj";
+
+    private static final String COMMA = ",";
+
+    public VariantVcfExacFactory() {
+        this(null);
+    }
+
+    /**
+     * @param tagMap Extends the VariantAggregatedVcfFactory(Properties properties) with two extra tags: Het and Hom.
+     * Example:
+     * <p>
+     * <pre>
+     * {@code
+     *
+     * SAS.AC = AC_SAS
+     * SAS.AN = AN_SAS
+     * SAS.HET=Het_SAS
+     * SAS.HOM=Hom_SAS
+     * ALL.AC =AC_Adj
+     * ALL.AN =AN_Adj
+     * ALL.HET=AC_Het
+     * ALL.HOM=AC_Hom
+     * }
+     * </pre>
+     * <p>
+     * Het is the list of heterozygous counts as listed by VariantVcfExacFactory.getHeterozygousGenotype() Hom is the
+     * list of homozygous counts as listed by VariantVcfExacFactory.getHomozygousGenotype()
+     */
+    public VariantVcfExacFactory(Properties tagMap) {
+        super(tagMap);
+    }
+
+    @Override
+    protected void loadMappings(Properties exacMappingProperties) {
+        super.loadMappings(exacMappingProperties);
+    }
+
+    @Override
+    protected void parseStats(Variant variant, String fileId, String studyId, int numAllele, String[] alternateAlleles,
+                              String info) {
+        VariantSourceEntry sourceEntry = variant.getSourceEntry(fileId, studyId);
+        VariantStatistics stats = new VariantStatistics(variant);
```

Comment:
I guess `VariantStats` and `VariantStatistics` are fully compatible?

Reply:
they have the same fields, yes